.gitlab-ci.yml

```diff
@@ -265,6 +265,32 @@ pkg:nix:
     paths:
       - nix-build-coq.drv-0/*/test-suite/logs
 
+build:macOS:
+  <<: *build-template
+  tags:
+    - macOS
+  before_script:
+    - brew update
+    - brew unlink python
+    - brew install opam gnu-time gtk+ expat gtksourceview gdk-pixbuf
+    - brew unlink python@2
+    - brew link python3
+    - pip3 install macpack
+    - opam init -a -y -j 2 --compiler=4.02.3 default https://opam.ocaml.org
```

Comment:
:+1: or at the minimum 4.05 to be consistent with the current setup for Windows packages.

Reply:
Windows packages use 4.06.1

Reply:
4.07 cannot be used because camlp4 is not available for that version under OPAM. So I'm trying 4.06.1.

Reply:
I am not sure camlp4 will be ever available again for newer OCaml versions, it has been declared unmaintained. Why does the OSX build need it while neither the Linux or Windows setup does?

Reply:
You are installing the wrong packages, in particular `lablgtk-extras`.

Reply:
>4.07 cannot be used because camlp4 is not available for that version under OPAM.

opam might install different things depending on the ocaml version used, don't worry about it unless it complains about unavailable packages. If it does it should tell you why it wants the package.

Reply:
Which also raises a question about duplication; I wonder if you could get the variables from the linux Docker image.

Reply:
> You are installing the wrong packages, in particular lablgtk-extras.

Thanks, I'll remove this one and see.

> Which also raises a question about duplication; I wonder if you could get the variables from the linux Docker image.

The Linux Docker image, I don't think so, but booting from a macOS docker image is certainly worth trying. I'd like to do it as a second step, though, since it will probably require tweaking the tool we use to make the `.app` bundle relocatable (macOS dark magic inside).

Reply:
> I wonder if you could get the variables from the linux Docker image.

Now I realize I misinterpreted your comment. Indeed in a setup with two Docker images, it would be interesting to know how to share variables. Maybe that's trivial with Docker?

Reply:
> Thanks, I'll remove this one and see.

You need to add `conf-gtksourceview.2` tho, see the dependencies.

Reply:
> Maybe that's trivial with Docker?

Certainly Docker does provide a way to have variables in Dockerfiles; indeed with our current image build setup is possible to use it.

Reply:
It was not used originally as we were using `hub.docker` to build images, and in that context variable instantiation was more problematic.

Reply:
I believe that the Windows packaging job still builds camlp4 (and does not use OPAM BTW, only the test-suite Windows test does).

Reply:
> Btw, does OPAM 2.0 come with official Windows support?

No.